app/src/main/cpp/skyline/services/am/applet/ILibraryAppletAccessor.h

```diff
@@ -5,21 +5,38 @@
 
 #include <kernel/types/KEvent.h>
 #include <services/serviceman.h>
+#include "IApplet.h"
+#include <applet/applet_creator.h>
 
 namespace skyline::service::am {
+    namespace result {
+        constexpr Result ObjectInvalid(128, 500);
+        constexpr Result OutOfBounds(128, 503);
+        constexpr Result NotAvailable(128, 2);
+    }
     /**
      * @brief ILibraryAppletAccessor is used to communicate with the library applet
      * @url https://switchbrew.org/wiki/Applet_Manager_services#ILibraryAppletAccessor
      */
     class ILibraryAppletAccessor : public BaseService {
       private:
+
         std::shared_ptr<kernel::type::KEvent> stateChangeEvent; //!< This KEvent is triggered when the applet's state changes
+        std::shared_ptr<kernel::type::KEvent> popNormalOutDataEvent; //!< This KEvent is triggered when there's normal data for the guest to pop
+        std::shared_ptr<kernel::type::KEvent> popInteractiveOutDataEvent; //!< This KEvent is triggered when there's interactive data for the guest to pop
+
+        KHandle stateChangeEventHandle{};
```

Comment:
Are you sure this is right? Are handles actually cache

Reply:
Ryujinx seems to do this
https://github.com/Ryujinx/Ryujinx/blob/43ebd7a9bbba0c1290a9e98b9224f0752627c400/Ryujinx.HLE/HOS/Services/Am/AppletAE/AllSystemAppletProxiesService/LibraryAppletCreator/ILibraryAppletAccessor.cs#L34